sklearn/metrics/ranking.py

```diff
@@ -719,3 +720,67 @@ def label_ranking_loss(y_true, y_score, sample_weight=None):
     loss[np.logical_or(n_positives == 0, n_positives == n_labels)] = 0.
 
     return np.average(loss, weights=sample_weight)
+
+
+def precision_at_k_score(y_true, y_score, sample_weight=None, n_tops=1):
+    """Compute the precision for n_tops scored labels
+
+    Count how many time the ``n_tops`` label with highest score are in the set
+    of true labels. One minus precision at 1 is also called one error.
+    It's averaged over the samples and can take a value between 0 and 1.
+    The best performance achievable could be below 1.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples, n_labels]
+        True binary labels in binary indicator format.
+
+    y_score : array, shape = [n_samples, n_labels]
+        Target scores, can either be probability estimates of the positive
+        class, confidence values, or binary decisions.
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights.
+
+    n_tops : int, optional
+        The number of labels on which the precision is computed.
+
+    Returns
+    -------
+    score : float
+
+    """
+    y_true = check_array(y_true, ensure_2d=False, accept_sparse='csr')
+    y_score = check_array(y_score, ensure_2d=False)
+    check_consistent_length(y_true, y_score)
+
+    y_true = csr_matrix(y_true)
+
+    y_type = type_of_target(y_true)
+    if y_type not in ("multilabel-indicator",):
+        raise ValueError("{0} format is not supported".format(y_type))
+
+    if y_true.shape != y_score.shape:
+        raise ValueError("y_true and y_score have different shape")
+
+    if n_tops < 1:
+        raise ValueError("Number of top labels must >= 1, got %s"
+                         % n_tops)
+
+    n_samples, n_labels = y_true.shape
+
+    y_true = csr_matrix(y_true)
+
+    if n_tops > 1:
+        row_top_k = partition(y_score, kth=n_labels - n_tops,
+                              axis=1)[:, -n_tops]
+    else:
+        row_top_k = np.max(y_score, axis=1)
+
+    # Here we take into account the ties
+    y_thresholded = csr_matrix(y_score >= row_top_k.reshape((-1, 1)))
```

Comment:
IMO this in not correct in the case where in a given row multiple highest scores are equal: this will selects all `scores > k_best_score`, which is different that selecting k scores.

For instance:
```
import numpy as np
from scipy.sparse import csr_matrix

scores = np.array([
    [0, 0.9, 0.9, 0.9],
    [0, 0.95, 0.9, 0.95],
    [0.1, 0.3, 0.2, 0.1]
    ])

n_tops = 2
n_samples, n_labels = scores.shape

row_top_k =np.partition(scores, kth=n_labels - n_tops, axis=1)[:, -n_tops]

row_top_k
Out: array([0.9 , 0.95, 0.2 ])

y_thresholded = csr_matrix(scores >= row_top_k.reshape((-1, 1)))
y_thresholded.todense()
Out:
matrix([[False,  True,  True,  True],
        [False,  True, False,  True],
        [False,  True,  True, False]])

# k=3 on first row, instead of 2
y_thresholded.todense().sum(axis=1)
Out:
matrix([[3],
        [2],
        [2]])
```